### ISSUE TITLE
Encrypted Message from OpenPGPjs isn't determined to be encrypted

### ISSUE THREAD

**Issue by amrosado** [SecurityInnovation/PGPy#162]
I have the following PGP Message Object created from OpenPGPjs:
```

"-----BEGIN PGP MESSAGE-----
Version: OpenPGP.js v2.3.3
Comment: http://openpgpjs.org

wcFMA5GE+lYMmJecARAAsncRFaiTAR3A6pvhi8lQm9JAIUyXBP0zN9YSu83A
iW4zO8RnSNJKSLr2u6ySdWLfMNLag3M5DYUK1VQx0/qJl4yc+EOcHsuF7Yb7
EQITIud3YAKqKcDRljp2El/2tuZJLwjqEXzrpYgvHo9hEhEblVfhRXl0q95J
IugkM/P07UIa7SM0QvQB44MORtNJr34qWFCW3Pij2ZyzbCETNetwWpE55dXc
iULDEEypnRn+4WDK/LKHQ6rFwlKha0R1AehwGTQaiVWCU7LklhO2QvCi10tn
G/TpsTErbxwgvl6BnL5EF2Qw8DcUMPGIrwZxcrcks+Uw4uroZjd1kEaBAy+O
uktW/yj/OgZi1LLpfneIIfSpwa3dN8bP2WD5m1NsitPTACvnTPsp/JoI6Al6
AqaOYiDq77Nc3XFhv9lexGrdI407P6vOQGGDLA6tTR3du88ycJNSLM/JrG0Y
4uUSgFR25Aq+t+BJ5E4HE5EG7awdxwW4/DcvbSoI3qMht2OKc691qDoTTt9d
SJI/HbYgUL7EpLvtqwOl/JkxVtiIvY1cp9hsvmBcN8rsidveT2d3DJJX4hWY
/cQJIv0AkDR3IVmiS+ycghk5XKcUVmyiOW/05qxdhk7uLMDrOI6LZVXyNIca
YTQ89iGBH13bfHJE6BfIpHeFuM7b5wMH1FhY68j1i7PSzgQB1IMxUC7JdPfT
VSKr6uT73iT+DyctuIqwAvvFePaX18UqQ/IWnVGOTthg+a4DP5ejNUBF7kfo
be4ilq9CJL/MMTLlP7ZI8/788ZJlomyXutcjy6zCmhVbGBO+1B2XDbTRxJ0f
QwYC0FvG3aC+16nXtQ1tB0JIiXpuxtYaO356sWItyyfEUOLNExlU3hOGY9DH
Xa5xy9jd9Ie7kgiKR6NKkUTN2O8ydzsm0gW0zlsal8LAMQKz23cyAJvFoBY+
ykECxzwLOEO1F2n9shxPOpHfkOevClsCFOt/2vVOaH0txmclsO4JvSZ7f+/k
evvIkZdeZlKgqLiFcSKTnbqOAHF3BwbyctBPydm1lTTRhwcJ3CsumEFY5JBE
8YI3cXrBLRR3gNjM/SddLDD6OlHldSfkSOF4uelj3U6VfL5HbRppirFZZuOu
f3yLQB6tjFGaAKBC5gXAfm6pqN8JUf+gts0OP4zuIUrjCKOjBGT+CF9ojd/q
3/eAFe8KCjpeREUatpi6jJzg+s1R+mks/lyTZFJzFvz2qpuswZgwkS1640s4
DuUqeR8nQrweUp6F1QK6lCMVaTsYQZbyZO+jogy/05foqf0WlfRKQs/v25Rh
CAqXkEdD+B4Ahg0+FmGyQBI9mBv10oQxGUQAApGyxC4mxRuFM3ZKOPJxe7hY
SwbsQml+o5Z27Ni5IP2oMgi8galBvurGns+AGc+e8VL/Slhx3r8S9Kjc/fFI
8sMevoeeZj1KTA6+R0YbDmCgcE9xxhX9ct0ZpGPRs4efQpcilXrV+txPRDoq
SrG9QqhjgFBGhNkOH8FJ3mAnkC+weAB7yJ5bGn9Jm/SJphXLTqXoGRK08mmx
/li9QiNpgtoMdXNhZD82mKTCFC56A4uPR+wlD+BfOdInUqCtJMS1a9lu7Sww
4e8RTZdP8bEe7M5MfouttxydWmUgoznTMjTmwyhLasQOwZWZM2iRhKfI0Bn+
XSi3oAMsmlx7vL+Dej7sUiRdDSK6VRZloDwPzyIGy+fNM8K3RZpHiSYEj4yK
Z0Tw1J45argIH31i9KkkEfp6mL8EXFlREydsVVYPYXUGHDx7BXJ14ipPqLoS
3+9x5y5fPyqP1SjXIe6keXG+4MvWiqBaNbEjgo8S9l93wkno1Eow54xlBaDE
3x2qCV5JsKkBaqdzGul0oEauGhrHfLLtYqwTElwCm9J8R4sQm9rdmoqDGKD1
oa7BOQFTuNCFNvaMUrITnvO8yG0GycWz3okEovfjfhqeGqzcq4EoAOp6b3B5
3V7Ud+Ys5t7D3bUOs23PhDqwCWKJc9tJRY9ZM1iB82sfIoMNtCytMH92HSc8
AuwMLpHzhDP36f/Q8hkLa8px7Se9sdxUjZU5r/miX+QKmDzxIFgMZnSGPJ89
R3b/2vtvRGPraeK0aONUhDNuHHHr1tu4+8gBvjV7NsFQaFoCu8A8uM7NnrZ/
L7mKtjFFWTZo+JRpMKaIuvVWXip3VJeb9ZQ0uMpYHiMoa5QW0WOLcUH7yfHO
KqLeeX8l3xL5+v4UFCadbOF+yHgccuEtgkMydA3reWBkv2V0xU/4wOISe8qW
+4uxwZa/kLRxTCA+mSCdKw5fUplvOcpiSBmCdB+PqHLn6jVH40UHt82S0T4p
MEQ8qBdA2SHHBx7VoEAWl5IEVSFcWtdA2W/YmULfWu/jdItV519S0hWaX8rT
OHmC7882qTnM1s1X+jgtblkkcDZYHq4CwLYOBjTuIjw0V3vJ2kgvSamjOlmZ
URbmD1xflP/Kyd0LF+vgSlaRSKTuaabmOTLxjZmOiM44nakY3Awa78ojtuAq
MaiPJRO/cJfmsn02Vd2BhZOJOs+PiyE3P8cHXF8dT2VvxmYruHozeVKdJHmQ
jtW8bx1gJuAQb6yzRutXlUgZFx20QxatarqTML/zH3eBdsxVl/diGt3F2+56
0WeXvvG10KvNjxfv9jAoIN3wgWwk3HvCKObkuNqUjlkeqA3PPFm1ZzkjSXKq
4XkakUFYr5pEX1Xtcm7LEl9x+HJEePJTE3E47lxmT1/CRG8OPxJ9Po2WQbUA
/T6LaBJK1LKH/1OneNdSetsDsDoodz7p7bmaNY+NsvVb2qSo6wSJzGQgoE6R
GTVylI8E4+H9SUQ6AmDpwIm2HsTceY32EA3kKWaLwGv/oBVg/m+mwNRL5y6+
iuOT5RC7fS0l3o/FBFrBkZg8Ak2i4q0zA27AwNmt0PjITRmV6bYfQhVWJ8f9
MyeQF1Iff9/6mtZyc+ZruAaflKvL9o/OxPg5vBVnY76iGf6S2FqDBypTf32J
pte6dDoCQk7+NVrsELu3CCrP0S8XIKc5xfgZxaFFZ0COj/Isczc0KQxSUhwU
wDieG2IYF/wlEqDWsHVIYNlCMZF7c8Il2RLqceCpp/kBrRXYMbb628bPIWYg
6JYuQASZ7XpaO4XEdI3qKheM3iu9ZEDOrlbxKVnqmmPKu+FLcN7s+KGbzrf9
3OwgLXTGGpoO0skpfjhHDVIfjFGMK1aMLuUwqNtUwBs16jk6T/7GfI2TN4lD
nyLIz0txJ/OWdMnsNFm3XsHLTEAn+bY/esDdgLPFpt7YSoN1qLXUzEwUafz1
G1LqIhkVfafZJgwgxPLd1rxMWO/dqsV5imvrJD39j2P/hok+IaoO2wQqG9Eh
AyP/0tSwlfERW3BqhyZTkMR9BSooMZk4vNZiOsKtvk1T6gE5l/o4fCAcx3UG
A4ynJ+oq7TQFadAK9KLwmZLjofp9uZtsnQ2rDW53G76IIsaRo+qpa2QhUJdd
JIE0ViIiaIyBbTd7BXjWfssLTYZ3Imr/7sE2lIYUBGrJlt1nM2hJE/xDhAJH
K6maGBUHRE74zH1VLkHy80R9NXACV57W8UMpWM0LA68a6R6MGfw3vjuEWAWa
RRgjmkrIhxOYfTnyNG1MdDtJFb8op0aTF8g+a0F1Z1yjc5MDevG0kNBhtGMb
tK4TJ1AT1RggZ76/7rXkfzR6mbY52Gsr4rTRh1E2jFTkSWiM9rJFlfJVHndm
vu+U3QNi5EFRqVbzuRtMXUTazn/gPgoOM0tH+x8yOa7Ne7RCbMINai2JTGFH
EZBpui/+mPAmNRvwSyodpYauGFftjR8r3WBylvHJOFW5g4MUAYUX1Uaib0h3
q4VPhQO5GkLVfYiHrGLdHYhOxcsT+61ikFU7Qux9W1p1HcTSGFzFFL/5YvSj
tfiwcoHDxNtU/Oz5hgdiWf71WkRDHvziDR2Wlx6xLwSpXas8oSZZRyLk+4Xg
5egOrM3WvuHiepBKQr7r9xTHN2JEU3+ChKxA7SFrw4A4VY8awrTAUlmvyavk
6FOZ4yfSEJiApKS3FTtfVjkHm8Q5Mmbd6lnrpbqa7kwz6QL062xOU18V/i6q
aughJJ/n3kG2R5HD2vJzWryT7XiX1BFAohbWx0x9LMURqNLcUapspdjPS2D3
wgr6WTRukVximz+Y3wIUHrWzP4m8XGWLx1WTr3w/6nyYs+mqqhMw8aJeB2NR
8YWXI55ti2hBTGP+fmFHEL2E3i2Vy0zUr3qIWp1miMNPs8YeYEAQ3Or3osey
DABzdV6Dax6TOJhqbTIiSzhElQqJ7uS1C6qYWF1OrWiCAUSP2MzuTluxJHgi
r4Rts377RxMeHVhUQ8YW6WbLWPuvJLXOgoEyt5Zhj7o87TydKlfi8TFA59n9
rK6pMPLKBXiPe6ERfLr4xHrweWbVAY9XdElfXOxDN/1xK3N6eOT3gjb0GWOc
YtJ698+mPlLsHygqjY4W6RwtTHOy8fXF5MddVYMsWND6ku3vXCC8Rc4ogiT4
+t04hhYtRvoMq0DJ+Nl2th+B6pdVv3/RLhTFY373t6SmPItBTJmntGx5yCIL
8DV7e3yil7MttSZEstd2qydJzkqeJFwQNVLj7cHakbzlt3sZ8xaqZWaFkwEa
KLFq8Sp83iumtpSnGtsPgppQzn+ZmbYrOO6m5wu0OCWlG8aQgXG0BmrbMmyh
tyU29I5hnXga3oWTkIu8rEJ2GCk/BvvBBtkMhk4OEzoFlmJOK/3jeQ25D5v7
mdaa6VMmR3QEdF52WUrFl1IghuhxUohtYK2ZzJ5Lzwqhnk/u6JT7RjAWl7le
lXY+bU3nMzt7MPScUU4OSkC7xvDZ1z86GgNiIi6Vd9rf25D16vXwHjOu2euA
TlRCsB7JyQop2iAonywJq9X+mIss0gnnG3rE1US263rxt5xXIL/ZWTEh2cs8
tgKS+sZcLrGiZ3gW8rWRqPpwmXj91eyAiC316YadgX7IkzVaLubGuekquPCU
VxVSrW9vHt0Gpnb0CghXzdgDpR2fnLkzlUyu6q3g0RM+PqpE5nFZ+Tdy1IiZ
LcAd80uOnTjSuIkoNtqNqKdoTp037Do0k2d6SoFe9ImaoM4+BZ5GCopRPZdg
7UN0kSGsahAR7+O6AHRQYq+mhidnbPf9+qwDaO7jhU/a7FyBY9ADht/DJjOQ
IrnGCjf5AqMYEXzmsKJYb2cZ2tuT+tmB63K03AdGhq+iBQ+Iuw841TyMoapm
DtQAdHL7qS5AVjWOVNuZ3oX1OmXbNXASoWiPACC1I4Jl/toKgZaFgSdJq++Y
qcYErUFvZ53n8uNIIr/BM3A2EbWjBojg43ZL7cmidxJN9lMczftlDImBgkXi
YmA0P9+LfKoJDwQQi4mg6OxMSvtEyAMMvzvv0tKJ1WKjGQxPtzol1hKdPaD3
7+K4FpM3vng2o4//e+aJjEOdt0M0niry3BcfYs/deM7jqY58dpuSw8GCxlaR
/Oo7FUYGsMUzNLXxj+78LOJe2i51RIxWebAMQ/f4oTia8AwS90++4yPzY7/Q
d3rw1BLOYEiTc0JYjipuwRC1Tpr0D8SwazxWaYAm+ya+Qs0unJ5anBuNMQ9p
esv982U19J6GLbKhmPlhnzNFS3nT+HmmjB4sPCqgBbWhX3YM9wNJNiKLYIYe
nnRhhiZ0LUPzt/YIyJBBbuN+/3dwugybG+EyRcYxhch10YKzzKdO2F9yUn3Y
sVTUvV1nToicWOn7jM7SYTGDMwROn8trLIhSICMoSgXRMMY=
=ly98
-----END PGP MESSAGE-----
```

Unfortunately the message isn't interpreted correctly by PGPy because although it is able to find the subkeys, etc. PGPy determines that the message is unencrypted.  I'm not sure what I am doing wrong.  I had to play around for a while to get PGPy to read the message generated by openpgpjs.  It seems like openpgpjs created symmetric messages it might be interpreted correctly, but the assymetric message isn't.  What do you guys think?  I would like to work on this problem, but might need some guidance as to what to look for in debugging, etc.

**Comment by Commod0re** [comment]
This is what I get with my first pass looking at it using PGPy 0.4.0:

```python
>>> import pgpy
>>> msg = pgpy.PGPMessage.from_file('i162.asc')
>>> msg.encrypters
{'9184FA560C98979C'}
>>> msg.is_encrypted
True
```

I get the same results when loading the message via `PGPMessage.from_blob` as well. What are you doing to decrypt it after this point? I don't see anything in the encrypted message that looks to be unsupported by PGPy. Assuming this is PGPy failing to do something right, I would guess that it lies either in the decrypted packets, or the key (and probably a case of the error raised not being as useful as it could be). I will replicate this scenario to try to pinpoint the issue.

Thanks for the issue report :)

**Comment by Commod0re** [comment]
So here's what I tried.

With openpgp.js, I created a keypair, encrypted a simple message, and wrote them to files:

```javascript
var openpgp = require('openpgp');
var fs = require('fs');

openpgp.initWorker({path: 'openpgp.worker.js'});

var keyoptions = {
    userIds: [{ name: 'T.K. JS', email: 'openpgp.js@email.com'}],
    numBits: 1024
};

openpgp.generateKey(keyoptions).then(function (key) {
    fs.writeFile('seckey.asc', key.privateKeyArmored.replace(/\r/g, ''));

    var msgoptions = {
        data: "Hello from OpenPGP.js!",
        publicKeys: openpgp.key.readArmored(key.publicKeyArmored).keys,
        privateKeys: key.key
    };

    openpgp.encrypt(msgoptions).then(function (encmsg) {
        fs.writeFile('emsg.asc', encmsg.data.replace(/\r/g, ''));
    });
});
```

and then ran this script using PGPy 0.4.0:

```python
#!/usr/bin/env python
import pgpy

if __name__ == '__main__':
    key, _ = pgpy.PGPKey.from_file('./seckey.asc')
    emsg = pgpy.PGPMessage.from_file('./emsg.asc')

    dmsg = key.decrypt(emsg)
    print(dmsg.message)
```

and it seems to work fine:

```
issue162 % python decrypt.py
decrypt.py:27: UserWarning: Message was encrypted with this key's subkey: DB566822994E6920. Decrypting with that...
  dmsg = key.decrypt(emsg)
Hello from OpenPGP.js!
```

 I removed the carriage returns on the javascript side to avoid issue #156 after seeing it in my first attempt at reproducing this.

Are you maybe still getting that issue? The fix for that hasn't made it into a release, yet

**Comment by amrosado** [comment]
I realized I was using an old version of PGPy that I had performed some workarounds to get to work for my application.  I used pip to get 0.4.0 and I was able to get that working with changing the hash timing issue that we previously discussed and using the python string replace method to get rid of the carriage returns that I previously modified the regular expression in the past code to work around.  It now seems to be working fine.  Thanks so much for your help.

**Comment by amrosado** [comment]
I think the enumeration based determination of the packet/message was actually quite important.  I haven't had the time to look at your new code because I was working on other projects all day, but I'm assuming you finally got around to working on that?  I was interested in helping you with solving some of the issues so that I could learn more about PGP.

**Comment by Commod0re** [comment]
Which issue was that? Unless it's related to #156 or #157 I probably haven't fixed it yet 8)

**Comment by amrosado** [comment]
The reason why the message wasn't being determined as encrypted I think might of been because it didn't use the enumeration within the message.  I remember when I was trying to figure out how to work past the problem I encountered it, but was hesitant to change it.  I wasn't sure if there was anything I needed to do.  I'm not really aware of what changed in PGPy from 0.3.0 to 0.4.0.  I was so happy to finally get the project I was working on further developed that I haven't spent time to appreciate the updates you've made.

I think the fix you suggest is better for #156.

**Comment by amrosado** [comment]
Hey, is it possible to send you my resume.  I'm interested in working with very good/talented developers and I can see from this package that you are very wise with python.  My e-mail is arosad2@protonmail.ch.  Security Innovation sounds like an interesting company.  I don't do any hacking, but I like theorizing about security and have worked on some interesting projects in the past.  Most of my programming is not in my resume.
